elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -702,11 +704,18 @@ public PersistentResource getRelation(String relation, String id) {
      * @return collection relation
      */
     public Set<PersistentResource> getRelation(String relationName) {
-        if (requestScope.getTransaction() != null && !requestScope.getPredicates().isEmpty()) {
-            final Class<?> entityClass = dictionary.getParameterizedType(obj, relationName);
-            final String valType = dictionary.getBinding(entityClass);
-            final Set<Predicate> filters = new HashSet<>(requestScope.getPredicatesOfType(valType));
-            return getRelation(relationName, filters);
+        // if pagination, then sorting with filters...
+        boolean hasFilters = !requestScope.getPredicates().isEmpty();
+        Set<Predicate> filters = null;
+        if (requestScope.getTransaction() != null && requestScope.getQueryParams().isPresent()) {
```

Comment:
Sure. I will remove the transaction check there. I was thinking that was the way things were being checked before.